python/src/aiconfig/editor/client/src/components/prompt/PromptActionBar.tsx

```diff
@@ -107,6 +110,7 @@ export default memo(function PromptActionBar({
                   style={{ overflowY: "auto" }}
                 >
                   <ModelSettingsRenderer
+                    model={getPromptModelName(prompt, defaultConfigModelName)}
```

Comment:
I looked at the `getPromptModelName` method in https://github.com/lastmile-ai/aiconfig/blob/1c68007b971896e74c31fb68917cff46d22bb769/python/src/aiconfig/editor/client/src/utils/promptUtils.ts#L34

Just to sanity check, the value of this doesn't quite "matter" (since it's the Hugging Face Task (`"Text Generation"`) and not the model itself (`"mistralai/Mistral-7B-Instruct-v0.1"`)). All that matters is that we have changed the HF task (which for most cases will be the actual model itself like `dall-e 2`) and that changed value will be detected by the key and cause that component to re-render the settings right?

Reply:
Ya, it's technically the model id that's registered in the ModelParserRegistry. And yep, that's correct (just nit on terminology, it would re-mount instead of re-render)

Reply:
This is beyond scope of this PR, just curious, what's the difference between re-mount vs. re-render and why does changing the key cause it to re-mount? Is Chat GPT wrong here? https://lastmileai.dev/workbooks/clsoufolp001pqrcsg97aj5dr

Reply:
Yep, it's wrong -- see https://react.dev/reference/react/useState#resetting-state-with-a-key

> You can reset a component’s state by passing a different key to a component. In this example, the Reset button changes the version state variable, which we pass as a key to the Form. When the key changes, React re-creates the Form component (and all of its children) from scratch, so its state gets reset.

Reply:
Re-mount basically means the component is removed from the DOM, recreated and re-added (state, etc. is re-initialized). Re-render just means updating the existing component (state, etc. is persisted)